### ISSUE TITLE
'sy.TorchHook(torch)' cause error

### ISSUE THREAD

**Issue by XiaotianJia** [OpenMined/PySyft#3986]
I use Pysyft to train the network, which involves the following operations:

```python
import torch
import numpy as np
import syft as sy

hook = sy.TorchHook(torch)
data = torch.rand(3, 2).to("cuda")
index1 = np.array([2,1,0,0,1,1,0,2,2,1,0,0,1,1,0,2,2,1,0,0,1,1,0,2,2,1,0,0,1,1,0,2])
index1 = torch.from_numpy(index1).long().to("cuda")
index2 = np.array([2,1,0,0,1,1,0,2])
index2 = torch.from_numpy(index2).long().to("cuda")
print(data[index1])
print(data[index2])
```

When `print(data[index2])` is executed, it causes the error “too many indices for tensor of dimension 2”. But when `hook = sy.TorchHook(torch)
` was commented out, the code runs successfully. Why is this happening? 

My environment:
pytorch: 1.4.0
syft: 0.25

**Comment by gmuraru** [comment]
Hmmm..if you remove the ```to("cuda")``` it works?

**Comment by XiaotianJia** [comment]
I removed the `to("cuda")`,  and it still doesn't work.

**Comment by gmuraru** [comment]
Yep - there is an issue. If you try to directly use ```index2``` without converting it to a tensor it works.
Thank you for signaling it :D

**Comment by XiaotianJia** [comment]
Yes. If `index2` is a numpy array, it works. Will the issue be resolved?

**Comment by gmuraru** [comment]
We will start looking into this.

**Comment by Linardos** [comment]
This issue should be reopened, I had the exact same problem and it was really hard to figure out it was a bug from PySyft:
![image](https://user-images.githubusercontent.com/26694607/106267656-a28d7b00-6232-11eb-8e28-64bdc4ae7063.png)
Note how the shapes are the exact same. What I was basicaly trying to do is define 3 separate masks based on the full mask of the heart. 
If I comment out the hook part (hook = sy.TorchHook(torch)) it works

**Comment by Saugatkafley** [comment]
As of `Jan 2022`, this error still remains , 
![image](https://user-images.githubusercontent.com/52364337/151281193-3eda6fad-de71-451b-a965-382fd9b236ef.png)
